tests/test_target_storage_types.py

```diff
@@ -0,0 +1,592 @@
+        create_info = [{'product': self.apple, 'qty': 2}]
+        picking = self.create_picking(self.picking_type_in,
+                                      products_info=create_info,
+                                      confirm=True)
+
+        self.validate_move_lines(picking.move_line_ids, self.apple, 2, 0)
+        self.picking_type_in.u_target_storage_format = 'package'
+        products_info = [{'product_barcode': self.apple.barcode, 'qty': 4}]
+        picking.update_picking(products_info=products_info, result_package_name=package.name)
+        self.validate_move_lines(picking.move_line_ids, self.apple, 2, 4,
+                                 is_package=True, package_name=package.name)
+
+    def test07_target_storage_format_package_over_receive_pallet_products(self):
+        """Tests over receiving for target_storage_format pallet_products"""
+        Package = self.env['stock.quant.package']
+        package = Package.get_package('test_package', create=True)
+        create_info = [{'product': self.apple, 'qty': 2}]
+        picking = self.create_picking(self.picking_type_in,
+                                      products_info=create_info,
+                                      confirm=True)
+
+        self.validate_move_lines(picking.move_line_ids, self.apple, 2, 0)
+        self.picking_type_in.u_target_storage_format = 'pallet_products'
+        products_info = [{'product_barcode': self.apple.barcode, 'qty': 4}]
+        picking.update_picking(products_info=products_info, result_package_name=package.name)
+        self.validate_move_lines(picking.move_line_ids, self.apple, 2, 4,
+                                 is_pallet=True, package_name=package.name)
+
+    def test08_target_storage_format_package_over_receive_pallet_packages(self):
+        """Tests over receiving for target_storage_format pallet_packages"""
+        Package = self.env['stock.quant.package']
+        package = Package.get_package('test_package', create=True)
+        create_info = [{'product': self.apple, 'qty': 2}]
+        picking = self.create_picking(self.picking_type_in,
+                                      products_info=create_info,
+                                      confirm=True)
+
+        self.validate_move_lines(picking.move_line_ids, self.apple, 2, 0)
+        self.picking_type_in.u_target_storage_format = 'pallet_packages'
+        products_info = [{'product_barcode': self.apple.barcode, 'qty': 4}]
+        picking.update_picking(products_info=products_info, result_package_name=package.name)
+        self.validate_move_lines(picking.move_line_ids, self.apple, 2, 4,
+                                 is_pallet=True, is_package=True, package_name=package.name)
+
+
+    def test09_target_storage_format_package_serial_numbers_product(self):
+        """Tests receiving tracked products for target_storage_format product"""
+        create_info = [{'product': self.strawberry, 'qty': 2}]
+        strawberry_sn = ['Strawberry0', 'Strawberry1']
```

Comment:
since this is not used in create_picking, I would move it after it and before products_info = ....

Reply:
Same for the other tests.